php/libraries/ProjectID.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types=1);
+/**
+ * A representation of a ProjectID object. A ProjectID is always an integer
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class ProjectID extends ValidatableIdentifier implements \JsonSerializable
+{
+    /**
+     * Returns this identifier type
+     *
+     * @return string
+     */
+    public function getType(): string
+    {
+        return 'ProjectID';
+    }
+
+    /**
+     * Validate that the value of the ProjectID is a positive integer that does
+     *
+     * This does not check for uniqueness in the database or any other
```

Comment:
So a ProjectID can still store a non-valid ID at the moment?

Reply:
Yes, that's true of all our ID classes. (This was based on SessionID which was based on CandID class). They validate the structure of the identifier but not the value. Database calls are expensive to be making as often as these classes are instantiated (especially if you're doing something like going through the results of `SELECT ProjectID, otherstuff from sometable` in order to implement the getProjectID function on a data model.)

Reply:
I understand and it's good to know.